### ISSUE TITLE
Can I get s3 output format like y=2016/m=01/d=25/h=17/M=00 (M=15,M=30,M=45)?

### ISSUE THREAD

**Issue by bernie4321** [fluent/fluentd#784]
I am being asked to log directly to s3. The requested s3 format would look like s3-bucket/logs/y=2016/m=01/d=25/h=17/M=00/ info.log-xxxx-x
where m=00, m=15, m=30, and m=45.

I can't think of a way of getting this without having four separate match/store blocks. Maybe I am missing something obvious with tags?

The file structure now is currently /mnt/logs/2016/01/25/info.log but it could even change on the file system to /mnt/logs/2016/01/25/h=17/M=15.

Thanks for your help.

**Comment by repeatedly** [comment]
> where m=00, m=15, m=30, and m=45.

There is no way for now.
If you want to do it, extending S3 plugin and change `@time_slicer` is better.
https://github.com/fluent/fluentd/blob/master/lib/fluent/output.rb#L516-L528
https://github.com/fluent/fluentd/blob/master/lib/fluent/output.rb#L565

**Comment by tagomoris** [comment]
`timekey_range` configuration parameter for Fluentd `v0.14` will solve this problem.

**Comment by repeatedly** [comment]
@tagomoris From the master, timekey_range is already supported. Can w close this issue?

**Comment by repeatedly** [comment]
Or close after release fluentd v0.14.0.pre2?

**Comment by tagomoris** [comment]
No. There's no way to instantiate plugins with new APIs.
It depends on #912 .

**Comment by repeatedly** [comment]
Ah, I see. Okay.

**Comment by tagomoris** [comment]
#912 was merged into master.
I know fluent-plugin-s3 doesn't support that feature yet, but it's the issue of that plugin.
